File: CS50/week2/readablity/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    double index, L = 0, S = 0, W = 0;
    string text = get_string("Text: ");
    for (int i = 0; i < strlen(text); i++)
    { // for how many letters

        if ((text[i] >= 'A' && text[i] <= 'Z') || (text[i] >= 'a' && text[i] <= 'z'))
        {
            L++;
        }
        //I don't how it works
        //  if (isalpha(text[i]))
        // {
        //     L++;
        // }
        // for how many sentences
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            S++;
        }
        // for how many words
        if (text[i] == ' ')
        {
            W++;
        }
        // For last char
        if ((text[i] == ',' || text[i] == '.' || text[i] == '?' || text[i] == '!') &&
            i == (strlen(text) - 1))
        {
            W++;
        }
    }

    L = (L / W) * 100 * 0.0588;
    S = (S / W) * 100 * 0.296;
    index = L - S - 15.8;
    index = round(index);
    if (index < 2)
    {
        printf("Before Grade 1");
    }
    else if (index > 16)
    {
        printf("Grade 16+");
    }
    else
    {
        printf("Grade %d", (int) index);
    }

    printf("\n");
}

File: CS50/week6/sentimental-readability/readability.py
import math

def main():
    L = 0
    S = 0
    W = 1  # 初始設為1，因為最後一個單詞不會被空格計數

    text = input("Text: ")

    for i in range(len(text)):
        # 計算字母數
        if text[i].isalpha():
            L += 1

        # 計算句子數
        if text[i] in ['.', '?', '!']:
            S += 1

        # 計算單詞數
        if text[i] == ' ':
            W += 1

    L = (L / W) * 100 * 0.0588
    S = (S / W) * 100 * 0.296
    index = L - S - 15.8
    index = round(index)

    if index < 1:
        print("Before Grade 1")
    elif index >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {index}")

if __name__ == "__main__":
    main()
